packages/react/src/settings/hooks/useConfiguration.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  fetchConfiguration as fetchConfigurationAction,
+  getConfiguration,
+  getConfigurationError,
+  isConfigurationFetched,
+  isConfigurationLoading,
+  StoreState,
+} from '@farfetch/blackout-redux';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction';
+import type { Configuration } from '@farfetch/blackout-client';
+import type { UseConfigurationOptions } from './types';
+
+const useConfiguration = (
+  configurationCode: Configuration['code'],
+  options: UseConfigurationOptions = {},
+) => {
+  const { enableAutoFetch = true, fetchQuery, fetchConfig } = options;
+
+  const error = useSelector((state: StoreState) =>
+    getConfigurationError(state, configurationCode),
+  );
+  const isLoading = useSelector((state: StoreState) =>
+    isConfigurationLoading(state, configurationCode),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isConfigurationFetched(state, configurationCode),
+  );
+  const configuration = useSelector((state: StoreState) =>
+    getConfiguration(state, configurationCode),
+  );
+  const fetch = useAction(fetchConfigurationAction);
+
+  const fetchConfiguration = useCallback(
+    () => fetch(configurationCode, fetchQuery, fetchConfig),
+    [fetch, configurationCode, fetchQuery, fetchConfig],
+  );
+
+  useEffect(() => {
+    if (!isLoading && !error && !isFetched && enableAutoFetch) {
```

Comment:
```suggestion
    if (!isLoading && !isFetched && enableAutoFetch && configurationCode) {
```

Reply:
I agree that `configurationCode` should be included as condition (as is a required required param of `fechConfiguration()` ). Shouldn't `!error` also be a condition?

Reply:
It is not necessary because for `isFetched` to be true one of 2 conditions need to be true:
 - It does have an error or it does have data.
 - It is not loading.
So when you include `!isFetched` in the if statement, you will only need to check that it is not loading to make sure that when `isFetched` is false it is due to not having an error or not having data. I am aware that other hooks still check for it explicitly (which does not hurt but is unnecessary). I will normalize them in a future PR.
Make sense?

Reply:
right, now it makes sense. `!isFetched` already covers the `!error` condition, I now see why it is not necessary. Thanks